packages/bpk-component-card-list/README.md

```diff
@@ -0,0 +1,41 @@
+# bpk-component-card-list
+
+> Backpack card list component.
+
+## Installation
+
+Check the main [Readme](https://github.com/skyscanner/backpack#usage) for a complete installation guide.
+
+## Usage
+
+### BpkCardList
+
+```js
+import { BpkCardList } from '@skyscanner/backpack-web/bpk-component-card-list';
+
+export default () => (
+  <BpkCardList
+    cardList={[]}
+    layoutDesktop="row"
+    layoutMobile="rail"
+    title="Card List Component"
+  />
+);
+```
+
+## Props
+
+### BpkCardList
+
+| Property            | PropType                               | Required | Default Value |
+| ------------------- | -------------------------------------- | -------- | ------------- |
+| accessory           | oneOf('expand' ,'button','pagination') | false    | -             |
+| buttonText          | string                                 | false    | -             |
+| cardList            | array(node)                            | true     | -             |
+| description         | string                                 | false    | -             |
+| expandText          | string                                 | false    | -             |
+| initiallyShownCards | number                                 | false    | 3             |
+| layoutDesktop       | oneOf('row','grid')                    | true     | -             |
+| layoutMobile        | oneOf('rail','stack')                  | true     | -             |
```

Comment:
Hey @FireRedNinja and @youngji0827 , I've been looking over this component as it's required for a piece of upcoming work in the Flights Booking tribe. One thing in the API that has struck out to me is the API props `layoutDesktop` and `layoutMobile`. This is a unique feature not repeated in any other Backpack component and I believe it's also not a common pattern in other UI libraries.

**What's the reason for doing it this way?** 

There are a number of concerns I have with this:

## Main concern - lack of flexibility
It couples any implementation to the internals of the component. 
  1. What if a design calls for a `stacked` or `rail` card list in desktop?
  1. What if I want to change display based on different breakpoints to those defined internally to the component?
  1. What if what defines the component I use isn't the width of the screen, but instead the width of the container that the card list is being placed in. An example here is for a modal or a drawer
  
More specifically of where this lack of flexibility hinders us is from designs I've been given for a new upcoming feature. Here the two "mobile" list layouts are needed on a large screen:

|  |  |
| ------------- | ------------- |
| Phase 1. A stacked card list on large screens | ![image](https://github.com/Skyscanner/backpack/assets/1532576/745b4677-ef0c-40a2-8820-922760a97913)  | 
| Future state. A rail card list on large screens | ![image](https://github.com/Skyscanner/backpack/assets/1532576/82c8cfb3-a3e2-4133-820d-c795ab3f9f2f)|

Instead of forcing these layouts to consumers, I would expect the decision on what layout to display to be decided by the parent, and the backpack component instead is dumb and renders what it's told to render.

## Minor concerns

### No support for SSR

`BpkBreakpoint` doesn't support rendering on the server, so there's a chance the traveller will see a flash of one style of component before the actual one appears. 

In most cases, how this should be handled is the microsite will use information from the request headers (e.g. user agent) to have a good guess at screen size, and thus render the layout that is appropriate. Locking this functionality to the internals of the component makes it impossible for the right layout to be rendered for the traveller when performing SSR.

This again is an argument for leaving the decision of what layout of the cards to the consumer of the component

### Bundlesize bloat

Coupling all 4 variants of cards together will inevitably lead to more JS and CSS being loaded to the page than is needed. One solution here again is to leave the decision of what layout to use up to the consumer. The consumer has many more tools available to limit the bundlesize (e.g. lazy loading), and thus is in a much better position to decide what to render.

I haven't looked into the specifics of the bundlesize, so if you have this information available it would be really useful to know!